crmprtd/process.py

```diff
@@ -196,16 +214,22 @@ def process(
             log.info(obs)
         return
 
+    obs_by_network_dict = obs_by_network(observations, sesh)
+
     log.info("Insert: start")
-    results = insert(
-        sesh,
-        observations,
-        strategy=insert_strategy,
-        bulk_chunk_size=bulk_chunk_size,
-        sample_size=sample_size,
-    )
-    log.info("Insert: done")
-    log.info("Data insertion results", extra={"results": results, "network": network})
+
+    for network_key in obs_by_network_dict:
```

Comment:
What you've done here reminds me of the Python standard library function [`itertools.groupby`](https://docs.python.org/3/library/itertools.html#itertools.groupby) (which would be similar to the SQL `GROUP BY` clause). Do you think that there's a way to utilize `itertools.groupby` to streamline what you have written here?